<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CwsServerConfig</class>
 <widget class="QDialog" name="CwsServerConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1212</width>
    <height>533</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>WMS Server Configuration</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_6">
   <item row="0" column="2">
    <widget class="QStackedWidget" name="m_pqWidgetStack">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="stackedWidgetPage1">
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>3</number>
       </property>
       <property name="bottomMargin">
        <number>3</number>
       </property>
       <item row="0" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Service Access</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="leftMargin">
           <number>3</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Application Server Port</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Maximum Thread Count</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Server User Password</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="m_pqleServerAdmin">
            <property name="text">
             <string/>
            </property>
            <property name="echoMode">
             <enum>QLineEdit::Password</enum>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="m_pqsbThreads"/>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="m_pqleServerUser">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Server User</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QSpinBox" name="m_pqsbMaxConnection"/>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="m_pqsbDaemonPort">
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QCheckBox" name="m_pqchbZipCompression">
            <property name="text">
             <string>Zip Compression</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Maximum Connection Count</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Maintenance Port</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="m_pqsbMaintenancePort">
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QSpinBox" name="m_pqsbSessionTimeout"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Session Timeout (Minutes)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>6</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="stackedWidgetPage2">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Daemon Konfiguration</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Primary Port Application Service</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Primary Port Maintenance Service</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Server Application Executable</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Secondary Port Maintenance Service</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Secondary Port Application Service</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="m_pqleApplicationExecutablePath">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QPushButton" name="pushButton_3">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Best Switching Hour End</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Best Switching Hour Start</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Process Health Check Intervall (minutes)</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Max Server Age (hours)</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_36">
            <property name="text">
             <string>Max Process MemoryUsage (MB)</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbPrimaryApplicationPort">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QCheckBox" name="m_pqchbSimulateProcessManagement">
            <property name="text">
             <string>Simulate Process Management</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbPrimaryMaintenancePort">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbSecondaryApplicationPort">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="5" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbProcessHealthCheckIntervall"/>
          </item>
          <item row="7" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbBestSwitching">
            <property name="maximum">
             <number>24</number>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbSecondaryMaintenancePort">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="6" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbMaxServerAge">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100000</number>
            </property>
           </widget>
          </item>
          <item row="8" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbBestSwitchingEnd">
            <property name="maximum">
             <number>24</number>
            </property>
           </widget>
          </item>
          <item row="9" column="1" colspan="2">
           <widget class="QSpinBox" name="m_pqsbMaxProcessMemoryUsage">
            <property name="maximum">
             <number>100000</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QCheckBox" name="m_pqchbSchemeBasedUserManagement">
            <property name="text">
             <string>Scheme Based UserManagement</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="m_pqgbDaemonServerLogging">
         <property name="title">
          <string>Daemon Logging</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="leftMargin">
           <number>3</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>Daemon Log Path</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="m_pqleDaemonLogPath"/>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="m_pqpbDaemonPathSelection">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_35">
            <property name="text">
             <string>Daemon Logging Severity</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QCheckBox" name="m_pqchbDaemonSeverityInfo">
              <property name="text">
               <string>Information</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbDaemonSeverityWarning">
              <property name="text">
               <string>Warning</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbDaemonSeverityError">
              <property name="text">
               <string>Error</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbDaemonSeverityFatal">
              <property name="text">
               <string>Fatal</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="m_pqgbApplicationServerLogging">
         <property name="title">
          <string>Application Server Logging</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Path</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="m_pqleApplicationLoggingPath"/>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="m_pqpbSelectApplicationLoggingPath">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Severity</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QCheckBox" name="m_pqchbApplicationSeverityInfo">
              <property name="text">
               <string>Information</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbApplicationSeverityWarning">
              <property name="text">
               <string>Warning</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbApplicationSeverityError">
              <property name="text">
               <string>Error</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbApplicationSeverityFatal">
              <property name="text">
               <string>Fatal</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>295</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_2">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="CwmsPluginsIf" name="m_pCwmsPlugins" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_3">
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="0" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Server Host Url for URI</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="m_pqleServerHost"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Server Url Port</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="m_pqsbServerPort">
         <property name="maximum">
          <number>100000</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>Server Url Path</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="m_pqleServerPath"/>
       </item>
       <item row="3" column="1">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>403</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="m_pqpbDataAccess">
       <property name="text">
        <string>Data Access</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="m_pqpbCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="m_pqpbOk">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="1">
    <widget class="QTreeWidget" name="m_pqtwOptions">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <column>
      <property name="text">
       <string>Options</string>
      </property>
     </column>
     <item>
      <property name="text">
       <string>Service Access</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Daemon Configuration</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Logging</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Plugins</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>URL Config</string>
      </property>
     </item>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CwmsPluginsIf</class>
   <extends>QWidget</extends>
   <header location="global">CwmsPluginsIf.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_pqpbDataAccess</sender>
   <signal>clicked()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>DataAccessClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>729</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbCancel</sender>
   <signal>clicked()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1120</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>624</x>
     <y>676</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbOk</sender>
   <signal>clicked()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>OKClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1201</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>818</x>
     <y>719</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqtwOptions</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>OptionSelectedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>168</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>631</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>ServerExecutableClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1176</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>1252</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbDaemonPathSelection</sender>
   <signal>clicked()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>DaemonLogPathClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>1255</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbSelectApplicationLoggingPath</sender>
   <signal>clicked()</signal>
   <receiver>CwsServerConfig</receiver>
   <slot>ApplicationLogPathClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>1236</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>DataAccessClickedSlot()</slot>
  <slot>OKClickedSlot()</slot>
  <slot>DaemonLogPathClickedSlot()</slot>
  <slot>ServerExecutableClickedSlot()</slot>
  <slot>ApplicationLogPathClickedSlot()</slot>
  <slot>OptionSelectedSlot()</slot>
 </slots>
</ui>
